gstaichi/runtime/program_impls/llvm/llvm_program.h

```diff
@@ -182,20 +183,46 @@ class LlvmProgramImpl : public ProgramImpl {
     runtime_exec_->check_runtime_error(result_buffer);
   }
 
-  size_t get_field_in_tree_offset(int tree_id, const SNode *child) override {
+  size_t get_field_in_tree_offset(int tree_id,
+                                  const SNode *child,
+                                  bool is_memory_aligned = false) override {
     // FIXME: Compute the proper offset. Current method taken from GGUI code
+    // Now fix the offset computation to support memory alignment ,
+    // the assumptions (That is, the parent of the snode is a dense, and the
+    // parent of that node is a root.) from GGUI code listed in field_info.cpp
+    // still holds
+
     size_t offset = 0;
 
     SNode *dense_parent = child->parent;
     SNode *root = dense_parent->parent;
 
     int child_id = root->child_id(dense_parent);
 
+    // The maximum cell size bytes is used to handle memory alignment for dlpack
+    // usage When compute field offset, SNode doesn't take the alignment into
+    // account yet, we need to handle it here
+
+    // The second ->ch is to get the actual data continer
```

Comment:
`data container`

Reply:
(marking resolved, since alexis already approved the pr)